### ISSUE TITLE
Missing .npz files

### ISSUE THREAD

**Issue by anshumitts** [OctoberChang/X-Transformer#7]
Hi,
I tried running X-Bert and downloaded datasets using the given script but as mentioned in the read me I did not get *.npz files or anyother file other than .txt. Can you please look into and let me know if I am not able to follow the instructions properly?
-AM

**Comment by smeaktrobush** [comment]
You can download preprocessed datasets from [here](http://manikvarma.org/downloads/XC/XMLRepository.html), remove the first line, and then run proc_data.py and label_embedding.py to get the .npz files.

**Comment by MartinaSoluncevska** [comment]
@smeaktrobush I encountered the same issue as @anshumitts. I tried the steps you proposed, but It fails to generate a L.elmo.npz file. I receive "unknown embed_type" in return.
Even if i continue without having this file, when running any prediction script file, i get 
  `AssertionError assert X.shape[1] == self.nr_features`

Did you maybe face any of this issues/found a solution for them?

**Comment by smeaktrobush** [comment]
@MartinaSoluncevska Apparently the authors have removed the ELMo embedding support (or maybe it wasn't included in the first place). You can try using 'pifa' instead of ELMo when generating label embedding.

**Comment by MartinaSoluncevska** [comment]
@smeaktrobush yup i tried and succeeded in generating the 'pifa' .npz file only. However, i still get the AssertionError and i can't proceed further :(

**Comment by smeaktrobush** [comment]
@MartinaSoluncevska can you give me more detail, i didn't face this problem.

**Comment by MartinaSoluncevska** [comment]
@smeaktrobush Here are the steps i followed: 
- downloaded pretrained model with given command `bash download-models.sh Eurlex-4K`
- downloaded "EURLex-4K" dataset from [here](http://manikvarma.org/downloads/XC/XMLRepository.html)
- changed stats to these `Eurlex-4K": [15449, 3865, 186104, 3956]` (the ones given by authors didn't match). Line 18, proc_data.py
- ran proc_data.py, which generated *.npz for train and test datasets.
- ran label_embedding.py, with 'pifa' embedding type.
- ran bash scripts/run_linear_eval.sh ${DATASETS} as given in instructions. This gives me the error.

**System information**
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 16.04
TensorFlow installed from (our builds, or upstream TensorFlow): 1.13.1
TensorFlow version (use command below):1.13
Python version: 3.6
Bazel version (if compiling from source): 0.19.2
CUDA/cuDNN version: 10.0 / 7.0

**Comment by smeaktrobush** [comment]
@MartinaSoluncevska I am assuming you are using the pretrained model. My guess is that since the data doesn't match, the features don't too. So maybe try training it from scratch?

**Comment by MartinaSoluncevska** [comment]
@smeaktrobush yeah, i will probably try that approach. Did you also train a model on your own or you used a pretrained one, since you mention you didn't face any errors?
Thanks for replies!

**Comment by smeaktrobush** [comment]
@MartinaSoluncevska No problem! To answer your question, I only trained it on a new dataset. Haven't tried the pretrained ones yet.

**Comment by MartinaSoluncevska** [comment]
@smeaktrobush Hi, sorry to be a bit persistent but i really want to give this project a try. Could you tell me something more about how to create all the dataset files needed, if you have your own data and not a public dataset? 
I don't really see how to create train.txt and text.txt in the file format needed, which seem like most important step at the beginning. Also, how do you get the numbers over [here](http://manikvarma.org/downloads/XC/XMLRepository.html) like feature & label dimensionality?

**Comment by smeaktrobush** [comment]
@MartinaSoluncevska You need to transform your data to [this](http://svmlight.joachims.org/) format to get train.txt, etc. You can get those numbers after you preprocessed it to tf-idf.

**Comment by Anastasiawang** [comment]
@smeaktrobush Hi，I have found the address of the training file in your answer above, but only the training file and test file in TF IDF sparse matrix format, and I didn't find the raw text file and label_vocab.txt. Where can I get them? In addition, I know in the preprocess.py file that raw test needs to be read, but in which .py file to read label_vocab.txt?

**Comment by smeaktrobush** [comment]
@Anastasiawang You can get the raw text file by running the download-data.sh script. But for the label_vocab.txt, I don't know either. If I had to guess, some datasets may not contain label text (so you will have to use 'pifa' embedding). For your last question, maybe it was used for getting ELMo embedding for labels however it was removed(?). I can't say much since I'm just a user of this repo too.

**Comment by anshumitts** [comment]
@smeaktrobush support for elmo was indeed removed as older version of this code did have elmo.
@Anastasiawang  This dataset simply downloads dataset from [here](http://manikvarma.org/downloads/XC/XMLRepository.html) Therefore, unlikely to have label_vocab you will have to create your own vocab file. you can use [pyxclib](https://github.com/kunaldahiya/pyxclib) for this purpose.
@MartinaSoluncevska You can also use [pyxclib](https://github.com/kunaldahiya/pyxclib) to get the appropriate format for test.txt and train.txt. Use header=False while using write_data.

**Comment by venomouscyanide** [comment]
@anshumitts Can you please tell me how to create my own vocab file using  pyxclib?
For context: I am trying to run this project for custom data that I have

**Comment by OctoberChang** [comment]
See how to run on the customized dataset https://github.com/OctoberChang/X-Transformer#running-x-transformer-on-customized-datasets
